packages/@react-stately/table/src/TableUtils.ts

```diff
@@ -133,42 +65,215 @@ export interface IColumn {
+    if (item.frozen) {
+      return acc + item.targetMainSize;
+    }
+    return acc + item.baseSize;
+  }, 0);
+
+  let remainingFreeSpace = initialFreeSpace;
+  // 9.7.4
+  // 9.7.4.a
+  while (flexItems.some(item => !item.frozen)) {
+    // 9.7.4.b
+    /**
+     * Calculate the remaining free space as for initial free space,
+     * above (9.7.3). If the sum of the unfrozen flex items’ flex factors is
+     * less than one, multiply the initial free space by this sum (of flex factors).
+     * If the magnitude of this value is less than the magnitude of
+     * the remaining free space, use this as the remaining free space.
+     */
+    remainingFreeSpace = availableWidth - flexItems.reduce((acc, item) => {
+      if (item.frozen) {
+        return acc + item.targetMainSize;
+      }
+      return acc + item.baseSize;
+    }, 0);
+    let flexFactors = flexItems.reduce((acc, item) => {
+      if (!item.frozen) {
+        return acc + item.flex;
+      }
+      return acc;
+    }, 0);
+    if (flexFactors < 1) {
+      let tempRemainingFreeSpace = initialFreeSpace * flexFactors;
+      if (tempRemainingFreeSpace < remainingFreeSpace) {
+        remainingFreeSpace = tempRemainingFreeSpace;
+      }
+    }
+    // 9.7.4.c
+    /**
+     * If the remaining free space is zero
+     * - Do nothing.
+     * Else // remember, we're always in grow mode
+     * - Find the ratio of the item’s flex grow factor to the
+     * sum of the flex grow factors of all unfrozen items on
+     * the line. Set the item’s target main size to its flex
+     * base size plus a fraction of the remaining free space
+     * proportional to the ratio.
+     */
+    if (remainingFreeSpace <= 0) {
+      // do nothing
+    } else {
+      let total = flexItems.reduce((acc, item) => item.frozen ? acc : acc + item.flex, 0);
+      flexItems.forEach((item) => {
+        if (item.frozen) {
+          return;
+        }
+        let ratio = item.flex / total;
+        item.targetMainSize = item.baseSize + (ratio * remainingFreeSpace);
+      });
+    }
+
+    // 9.7.4.d
+    /**
+     * Fix min/max violations. Clamp each non-frozen item’s
+     * target main size by its used min and max main sizes
+     * and floor its content-box size at zero. If the item’s
+     * target main size was made smaller by this, it’s a max
+     * violation. If the item’s target main size was made
+     * larger by this, it’s a min violation.
+     */
+    let totalViolation = 0;
+    flexItems.forEach(item => {
+      item.violation = '';
+      if (!item.frozen) {
+        let {min, max, targetMainSize} = item;
+        item.targetMainSize = Math.max(min, Math.min(targetMainSize, max));
+
+        totalViolation += item.targetMainSize - targetMainSize;
+        // got smaller
+        if (item.targetMainSize < targetMainSize) {
+          item.violation = 'max';
+        } else if (item.targetMainSize > targetMainSize) {
+          item.violation = 'min';
+        }
```

Comment:
You could avoid these strings by doing 

```js
item.violation = item.targetMainSize - targetMainSize;
totalViolation += item.violation;
```

And later use the sign of `item.violation`